### DISCUSSION TITLE
Switching from ChartFX 11.1.5 to 11.2.2

### DISCUSSION THREAD

**Discussion by asmirn1** [GSI-CS-CO/chart-fx#405] [Q&A] [answered]
Hi,

I've switched from ChartFX v11.1.5 to v11.2.2, and I've noticed a few differences in the charts. One of the differences is the axis labels ends with "[]" in the new version.

This is my chart built with version 11.1.5:
<img width="558" alt="11 1 5" src="https://user-images.githubusercontent.com/13631120/101920559-f4524700-3b99-11eb-890e-2d4dcbadb2cb.png">

And this is my chart built with version 11.2.2:
<img width="556" alt="11 2 2" src="https://user-images.githubusercontent.com/13631120/101920671-164bc980-3b9a-11eb-9342-346867f01b05.png">

And this is a part of my code (`RetTimeIntensityChart` extends `XYChart`)
```java
public class RetTimeIntensityChart extends DefaultChart {

    private final SortedMap<Double, Double> totalIntensitiesBuffer = new TreeMap<>();

    private final ErrorDataSetRenderer renderer;
    private final ErrorDataSetRenderer darkRenderer;

    private boolean showTotalIntensity = false;
    private boolean showSpectralFeatures = false;
    private FeatureList spectralFeatureList = null;


    public RetTimeIntensityChart() {
        super(new DefaultNumericAxis("Retention time, min"), new DefaultNumericAxis("Intensity"));
        this.setToolBarSide(Side.RIGHT);
        this.setToolBarPinned(true);
        this.setAnimated(false);
        this.getRenderers().clear();

        renderer = new ErrorDataSetRenderer();
        renderer.setPolyLineStyle(LineStyle.NORMAL);
        renderer.setErrorType(ErrorStyle.NONE);
        darkRenderer = new ErrorDataSetRenderer();
        darkRenderer.setDrawMarker(false);
        darkRenderer.setPolyLineStyle(LineStyle.NORMAL);
        darkRenderer.setShowInLegend(false);
        this.getRenderers().setAll(renderer, darkRenderer);
    }
```

**Comment by RalphSteinhagen** [comment, marked as answer]
@asmirn1 yes the square brackets '[]' are on purpose and contain the axis unit. If it shows as '[]' then the axis unit has not been set.

This is needed to automatically adjust the unit according to the axis range based on SI standards. Ie. if you set the auto-unit property and use, for example, 'V' as axis unit, then the unit string is automatically adjusted and plotted as '[uV]', '[mV]', '[V]', '[kV]', '[MV]' if the range becomes very large or small. This optional feature helps to avoid silly labels such as '0.000001 s'  rather than the more natural of '1 us' or to display '1200000000 eV' as '1.2 GeV'.

The auto-unit scaling isn't enabled by default though and the unit brackets can be suppressed by explicitly setting the axis unit to `null`.

Hope this helps and puts your valid observation into context a bit better.

**Comment by asmirn1** [comment]
@RalphSteinhagen Thank you for a detailed explanation. I've fixed the labels.

**Comment by RalphSteinhagen** [comment]
OK. I will close this issue. Feel free to re-open if this hasn't solved your problem.

**Comment by thhart** [comment]
Maybe I  miss something but how to hide the name and unit section at all? I can not get rid of the brackets at all?
Why the empty brackets are displayed at all? Wouldn't it be better if these would be hidden if unit is not set?

**Comment by RalphSteinhagen** [comment]
@thhart you may want to set the respective properties to 'null'. This should suppress the units and/or name.
